internal/pkg/cli/pipeline_init.go

```diff
@@ -277,16 +318,20 @@ func (o *initPipelineOpts) selectGitHubURL() error {
 
 	// Prompts user to select a repo URL.
 	url, err := o.prompt.SelectOne(
-		pipelineSelectGitHubURLPrompt,
-		pipelineSelectGitHubURLHelpPrompt,
+		pipelineSelectURLPrompt,
+		pipelineSelectURLHelpPrompt,
 		urls,
 	)
 	if err != nil {
-		return fmt.Errorf("select GitHub URL: %w", err)
+		return fmt.Errorf("select URL: %w", err)
+	}
+	if err := o.validateURL(url); err != nil {
+		return err
 	}
 	o.repoURL = url
 
 	return nil
+
 }
 
 func (o *initPipelineOpts) parseOwnerRepoName(url string) (string, string, error) {
```

Comment:
Might be better to add examples for those parsing methods. Otherwise it is a bit challenging to follow.

Reply:
Done.